lite2/client.go

```diff
@@ -454,6 +459,45 @@ func (c *Client) TrustedValidatorSet(height int64, now time.Time) (*types.Valida
 	return c.trustedStore.ValidatorSet(height)
 }
 
+// updateMissingTrustedHeaderAndVals finds the previous trusted height and then uses the sequence method to verify
+// all the headers at each height up to the inputted height.
+// If all are verified updates the store and returns the header and validator set.
+func (c *Client) updateMissingTrustedHeaderAndVals(height int64) (*types.SignedHeader, *types.ValidatorSet, error) {
+	var (
+		prevTrustedHeight int64
+		newHeader         *types.SignedHeader
+		newVals           *types.ValidatorSet
+		err               error
+	)
+	prevTrustedHeight, err = c.previousTrustedHeight(height)
+	if err != nil {
+		return nil, nil, err
+	}
+	c.trustedHeader, err = c.trustedStore.SignedHeader(prevTrustedHeight)
+	if err != nil {
+		return nil, nil, err
+	}
+	c.trustedNextVals, err = c.trustedStore.ValidatorSet(prevTrustedHeight)
+	if err != nil {
+		return nil, nil, err
+	}
+	newHeader, err = c.primary.SignedHeader(height)
+	if err != nil {
+		return nil, nil, err
+	}
+	newVals, err = c.primary.ValidatorSet(height)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// updates store if valid
+	err = c.sequence(newHeader, newVals, time.Now())
```

Comment:
this seems incomplete. we should do either a) bisection/sequence depending on verificationMode IF we're going forward b) `backwards` verification IF we're going backwards

Reply:
What is the criteria for deciding whether we are going forwards or backwards?